src/Compilers/Core/Portable/Collections/OrderPreservingMultiDictionary.cs

```diff
@@ -159,9 +163,9 @@ public struct ValueSet : IEnumerable<V>
             /// </summary>
             private readonly object _value;
 
-            internal ValueSet(V value)
+            internal ValueSet([DisallowNull] V value)
```

Comment:
>[DisallowNull] V value [](start = 30, length = 22)

If we can only use not null values, should we have a `where V : notnull` constraint?

Reply:
Good point. Let me try both variations out.

---
In reply to: [353925414](https://github.com/dotnet/roslyn/pull/40029#discussion_r353925414) [](ancestors = 353925414)